rcp/org.testeditor.rcp4.views.teststepselection/src/org/testeditor/rcp4/views/teststepselector/TestStepSelector.xtend

```diff
@@ -79,70 +96,94 @@ class TestStepSelector {
 
 				})
 		]
-		val page = PlatformUI.workbench.activeWorkbenchWindow.activePage;
-		page.addPartListener(new IPartListener() {
-
-			override partActivated(IWorkbenchPart part) {
-				if (part instanceof ITextEditor) {
-					refreshView(null)
+		
+		Job.getJobManager().addJobChangeListener(new JobChangeAdapter() {
+			override done(IJobChangeEvent event) {
+				if (event.job.name.equals("Building workspace")) {
+					logger.info("Building workspace completed. Trigger update TestSetSelector")
+					broker.post(TestStepSelector.SELECTOR_UPDATE_MODEL_AND_VIEW, null)
 				}
-			}
-
-			override partBroughtToTop(IWorkbenchPart part) {
-			}
-
-			override partClosed(IWorkbenchPart part) {
-			}
-
-			override partDeactivated(IWorkbenchPart part) {
-			}
-
-			override partOpened(IWorkbenchPart part) {
-			}
-
+			} 
 		})
 
 	}
 
-//	@Inject
-//	@Optional
-//	def void startupComplete(@EventTopic(UIEvents.UILifeCycle.APP_STARTUP_COMPLETE) Object data) {
-//		// startup complete ensures the index to be populated
-//		Display.getDefault.syncExec[populateViewIfEmpty] // is only run, if view is created on startup
-//	}
-
 	@Focus
 	def void setFocus() {
 		viewer.control.setFocus
 	}
 
 	@Inject
 	@Optional
-	def void refreshView(@EventTopic(SELECTOR_TOPIC_REFRESH) Object data) {
-		populated = false
-		logger.debug("Refreshing view.")
-		Display.getDefault.syncExec[populateViewIfEmpty]
+	def void updateModelAndView(@EventTopic(SELECTOR_UPDATE_MODEL_AND_VIEW) Object data) {
+		logger.debug("updateModelAndView")
+		updateModel
+		Display.getDefault.syncExec[internalUpdateView()]
 	}
 
-	// make sure that the index is populated before calling this method
-	// and that it is run in the swt thread!
-	private def populateViewIfEmpty() {
-		if (populated) {
-			return
+	@Inject
+	@Optional
+	def void updateView(@EventTopic(SELECTOR_UPDATE_VIEW) Object data) {
+		logger.debug("updateView")
+		// TODO - check on the project of the editor
+		if(viewer.input == null){
+			Display.getDefault.syncExec[internalUpdateView()]
 		}
-		val amlInjector = amlInjectorProvider.get
+	}
+
+	private def internalUpdateView() {
+
+		//Initial call to the selector
+		if (viewer.input == null) {
+			viewer.labelProvider = labelProvider
```

Comment:
Can we set the label and content provider in the post construct method? If yes, I think we don't need this case distinction here.